Update MakeUserCommand to make it inline fillable

alvleont opened the pull request:
This makes the artisan command make:filament-user inline fillable by using

<img width="575" alt="Captura de pantalla 2022-11-30 a la(s) 4 49 00 p m" src="https://user-images.githubusercontent.com/47951932/204915342-d897bd3d-f775-4afb-96a5-66bfcc183b70.png">

The PR also includes the --help options description and the options can be non-used inline: in that case, each option not used there will be prompted.

alvleont committed:
Update MakeUserCommand.php

zepfietje commented:
How does this handle input validation errors?

Reference from a past attempt: https://github.com/filamentphp/filament/pull/4414#issuecomment-1272297197.

alvleont commented:
Nice Question... The thing is: the method **validateInput** is the one making the eternal loop. One option (i think the more useful one) is to add an option to validateInput in order to define an additional behavior in case of error input. Since this is also Filament code, I'll try to add an additional modification there.

zepfietje commented:
Yeah would need to be updated to not loop infinitely. Mark this PR as ready for review when you've done so.

alvleont commented:
Working on it!😊

alvleont committed:
Changes in MakeUserCommand and CanValidateInput to get the inline working and keeping everything working as usual

Created a fallback variable for validateInput which is optional.

Applied to the Command.

alvleont committed:
Update MakeUserCommand.php

Delete validator, It's not used here.

alvleont committed:
Update MakeUserCommand.php

Sorry. Just me being stupid.

Deleted unnecessary code.

alvleont commented:
Done. Solution applied as proposed before.

alvleont commented:
😊😊😊 do I need to do something else?

danharrin commented on packages/admin/src/Commands/MakeUserCommand.php:
```diff
@@ -17,14 +17,18 @@ class MakeUserCommand extends Command
-            'name' => $this->validateInput(fn () => $this->ask('Name'), 'name', ['required']),
-            'email' => $this->validateInput(fn () => $this->ask('Email address'), 'email', ['required', 'email', 'unique:' . $this->getUserModel()]),
-            'password' => Hash::make($this->validateInput(fn () => $this->secret('Password'), 'password', ['required', 'min:8'])),
+            'name' => $this->validateInput(fn () => $this->options['name']??$this->ask('Name'), 'name', ['required'], function () { unset($this->options['name']);}),
+            'email' => $this->validateInput(fn () => $this->options['email']??$this->ask('Email address'), 'email', ['required', 'email', 'unique:' . $this->getUserModel()], function () { unset($this->options['email']);}),
+            'password' => Hash::make($this->validateInput(fn () => $this->options['password']??$this->secret('Password'), 'password', ['required', 'min:8'], function () { unset($this->options['password']);})),
```
/wtd Correct spacing between the null operator. Convert the closures to short closures.

danharrin committed:
Update CanValidateInput.php

danharrin committed:
Update packages/admin/src/Commands/MakeUserCommand.php

Co-authored-by: what-the-diff[bot] <114527992+what-the-diff[bot]@users.noreply.github.com>

danharrin committed:
Update MakeUserCommand.php

danharrin commented on packages/admin/src/Commands/MakeUserCommand.php:
```diff
@@ -17,14 +17,16 @@ class MakeUserCommand extends Command
     protected $description = 'Creates a Filament user.';
 
-    protected $signature = 'make:filament-user';
+    protected $signature = 'make:filament-user {--name= : The name of the user} {--email= : A valid and unique email address} {--password= : The password for the user (min. 8 characters)}';
```
/wtd split signature onto multiple lines and align the options

danharrin committed:
Update packages/admin/src/Commands/MakeUserCommand.php

Co-authored-by: what-the-diff[bot] <114527992+what-the-diff[bot]@users.noreply.github.com>

danharrin commented on packages/admin/src/Commands/MakeUserCommand.php:
```diff
@@ -17,14 +17,19 @@ class MakeUserCommand extends Command
-            'name' => $this->validateInput(fn () => $this->ask('Name'), 'name', ['required']),
-            'email' => $this->validateInput(fn () => $this->ask('Email address'), 'email', ['required', 'email', 'unique:' . $this->getUserModel()]),
-            'password' => Hash::make($this->validateInput(fn () => $this->secret('Password'), 'password', ['required', 'min:8'])),
+            'name' => $this->validateInput(fn () => $this->options['name'] ?? $this->ask('Name'), 'name', ['required'], fn () => unset($this->options['name'])),
+            'email' => $this->validateInput(fn () => $this->options['email'] ?? $this->ask('Email address'), 'email', ['required', 'email', 'unique:' . $this->getUserModel()], fn () => unset($this->options['email'])),
+            'password' => Hash::make($this->validateInput(fn () => $this->options['password'] ?? $this->secret('Password'), 'password', ['required', 'min:8'], fn () => unset($this->options['password']))),
```
/wtd Fix `unexpected token "unset"`

danharrin committed:
Update packages/admin/src/Commands/MakeUserCommand.php

Co-authored-by: what-the-diff[bot] <114527992+what-the-diff[bot]@users.noreply.github.com>

danharrin commented:
Thanks
